src/view/new-address.js

```diff
@@ -52,7 +52,13 @@ const NewAddressView = ({ store, nav, invoice }) => (
         >
           {store.walletAddress}
         </CopyButton>
-        <GlasButton onPress={() => nav.goHome()}>Done</GlasButton>
+        <GlasButton
+          onPress={() =>
+            store.syncedToChain ? nav.goHome() : nav.goLoaderSyncing()
+          }
```

Comment:
We should keep all logic outside of react components since it isn't really testable. The info action might be a good place for this.